### PR TITLE
Feat: improve iOS StackTrace

### LINKED ISSUE #42: Format Objective-C StackTrace.
Currently, Xamarin Grabs the Objective-C Exception message, stack trace, type and stores it into the C# Exception message.
By doing that, you'll not be able to Symbolicate the Exception and Grouping has the potential to mix Exceptions.

The Ideal Scenario is to Parse the Objective-C Exceptions with a new StackTraceProcessor for iOS by joining the C# StackTrace with the Native StackTrace.

For this to be possible, https://github.com/getsentry/sentry-dotnet/pull/744 needs to be merged in order to filter the Instruction Addresses from the Native Exception.

##Notes

Regex Rule for filtering the StackTrace:
`  => "(?<id>\\d+)\\s+(?<method>[a-zA-Z\\.-_?]+)\\s+(?<offset>0x[0-9a-fA-F]+)\\s+(?<function>.+?(?=\\s\\+))\\s+\\+\\s+(?<line>\\d+)";
`

### PR DESCRIPTION
**Pull request by lucas-zimerman**
**Before**
Before

**After**
After

This PR Aims to Close #42 by formatting Objective C Exceptions, and, also adding additional InAppExclude so that the final StackTrace gets less polluted by wrong InApp Frames.

I'm opening this as a draft because the UI of the StackTrace looks different compared to a StackTrace generated by the Cocoa SDK, I haven't found the root cause yet but It might be related to the fact that the StackTrace is attached to a Thread and not a raw StackTrace.

Sample Events
Before: https://sentry.io/organizations/sentry-sdks/issues/2174428307
After: https://sentry.io/organizations/sentry-sdks/issues/2174421919
Cocoa Exception: https://sentry.io/organizations/sentry-sdks/issues/2010891016

### CODE REVIEW

File: Samples/Sample.Xamarin.Core/ViewModels/MainPageViewModel.cs [hunk 1]
```diff
@@ -28,6 +28,12 @@ public MainPageViewModel()
             UnhandledCmd = new Command(Unhandle);
             BrokenViewCmd = new Command(GotoBrokenView);
             FeedbackCmd = new Command(ShowFeedback);
+            NativeCrashCmd = new Command(NativeCrash);
+            try
+            {
+                _nativeCrashService = DependencyService.Get<INativeCrash>();
+            }
+            finally { }
```
Comment (bruno-garcia) [inline Samples/Sample.Xamarin.Core/ViewModels/MainPageViewModel.cs:36]: why an empty finally block?

File: Src/Sentry.Xamarin/Internals/NativeExceptionProcessor.ios.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        private NativeStackTraceFactory _nativeStack;
+
+        public NativeExceptionProcessor(SentryXamarinOptions options)
+        {
+            _nativeStack = new NativeStackTraceFactory(options);
+        }
+
+        public void Process(Exception exception, SentryEvent sentryEvent)
+        {
+            if (exception is MonoTouchException monoException)
```
Comment (bruno-garcia) [inline Src/Sentry.Xamarin/Internals/NativeExceptionProcessor.ios.cs:19]: Cool, once we get `net6-ios` we can add this there

File: Src/Sentry.Xamarin/Internals/NativeStackTraceFactory.ios.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        public NativeStackTraceFactory(SentryXamarinOptions options) => _options = options;
+
+        internal bool IsNativeException(string exceptionValue)
+            => exceptionValue.StartsWith("Objective-C exception");
+
+        internal SentryStackTrace CreateNativeStackTrace(SentryStackTrace managedStackTrace, IEnumerable<string> nativeStackTrace)
+        {
+            var nativeFramesList = new List<SentryStackFrame>();
+            for (int i = nativeStackTrace.Count() - 1; i >= 0; i--)
```
Comment (bruno-garcia) [inline Src/Sentry.Xamarin/Internals/NativeStackTraceFactory.ios.cs:25]: You'll do multiple enumerations like this. Maybe take `ICollection<>` as an argument or `.ToArray()` in the method body to work with an array

File: Src/Sentry.Xamarin/Internals/NativeStackTraceFactory.ios.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            {
+                var match = Regex.Match(nativeStackTrace.ElementAt(i), _nativeRegexFormat);
+                if (match.Success)
+                {
+                    var method = match.Groups["method"].Value;
+                    nativeFramesList.Add(new SentryStackFrame()
```
Comment (bruno-garcia) [inline Src/Sentry.Xamarin/Internals/NativeStackTraceFactory.ios.cs:31]:
```suggestion
                    nativeFramesList.Add(new SentryStackFrame
```

File: Src/Sentry.Xamarin/Internals/NativeStackTraceFactory.ios.cs [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                    {
+                        Platform = "cocoa",
```
Comment (bruno-garcia) [inline Src/Sentry.Xamarin/Internals/NativeStackTraceFactory.ios.cs:33]: I don't think `platform=cocoa` is valid here. Isn't this supposed to be `native?
```suggestion
                        Platform = "native",
```

File: Src/Sentry.Xamarin/Sentry.Xamarin.csproj [hunk 6]
```diff
@@ -40,7 +40,7 @@
 		<None Include="Internals\*.ios.cs" />
 		<None Include="Internals\*.droid.cs" />
 		<None Include="Internals\*.uwp.cs" />
-	</ItemGroup>
+	</ItemGroup>
```
Comment (bruno-garcia) [inline Src/Sentry.Xamarin/Sentry.Xamarin.csproj:43]:
```suggestion
	</ItemGroup>
```

**Review by bruno-garcia — APPROVED** [review]
Some small improvements but LGTM

File: Src/Sentry.Xamarin/Internals/NativeStackTraceFactory.ios.cs (hunk 5 above)
Comment (lucas-zimerman) [reply Src/Sentry.Xamarin/Internals/NativeStackTraceFactory.ios.cs:33]: I'll change to native.
I was testing different names like cocoa and iOS (found it here https://github.com/getsentry/sentry-cocoa/blob/4e1b2fa5df554ddc9e30ca405b25fda89375e796/Tests/SentryTests/Protocol/TestData.swift#L122 )
but it seems like the result at the same output at Sentry UI.

File: Samples/Sample.Xamarin.Core/ViewModels/MainPageViewModel.cs (hunk 1 above)
Comment (lucas-zimerman) [reply Samples/Sample.Xamarin.Core/ViewModels/MainPageViewModel.cs:36]: To ignore  Platforms that didn't implement INativeCrash.

```csharp
        private Action NativeCrash => () =>
        {
            _nativeCrashService?.BrokenNativeCallback();
            DisplayAlert("Whoops", "A native crash sample wasn't implemented for this platform, mind opening a pull request? :)", "Yes");
        };
```

The Native Crash will not crash and an alert will be shown to the user explaining it.

Comment (bruno-garcia) [reply Samples/Sample.Xamarin.Core/ViewModels/MainPageViewModel.cs:36]: shou;dn't it be a catch block then?

File: Src/Sentry.Xamarin/Internals/NativeStackTraceFactory.ios.cs (hunk 5 above)
Comment (bruno-garcia) [reply Src/Sentry.Xamarin/Internals/NativeStackTraceFactory.ios.cs:33]: `native` can trigger symbolication, so if you have symbols, not sure if we upload it it'll work already or if it needs more data

File: Samples/Sample.Xamarin.Core/ViewModels/MainPageViewModel.cs (hunk 1 above)
Comment (lucas-zimerman) [reply Samples/Sample.Xamarin.Core/ViewModels/MainPageViewModel.cs:36]: whoops, it seems like in that case, it doesn't throw an exception> Removing the try/finally block.

**Review by bruno-garcia — APPROVED** [review]

**Pull request merged**